[#4] Add a background color option

ppuleo opened the pull request:
- Add a background color option
- Document the existing options
- Update the demo to use all options

ppuleo commented:
Hi @stewartlord, here's an idea for allowing the background color to be set as an option. If other options are to be exposed, maybe an options object would make sense as the constructor argument instead of a list? I'm happy to do that if you prefer.

stewartlord commented:
Hey Phil, so sorry for the delay. I just had a baby so I'm a bit overwhelmed at the moment. I like the idea of exposing more options as a hash if you are still willing to do that. We could keep hash as a standard argument since it is the only required param.

ppuleo commented:
Hi @stewartlord, no problem and congratulations! I'll update the PR with the options hash approach.

stewartlord commented:
Hi Phil, just cleaning up and I came across this PR. I was wondering if you wanted to keep it open?

ppuleo commented:
Hi @stewartlord, sorry I meant to update this and resubmit. I'll revisit it this morning.

ppuleo commented:
Hi @stewartlord, I've updated the PR with the changes we discussed plus a couple of other changes I've been using:
1. Updated the margin calculation to ensure the cells are always centered in the image.
2. If no hash is specified, create a reasonably randomized hash from the current time.
3. Optionally accept the size and margin arguments instead of the options object argument so the previous constructor signature still works for backward compatibility. 

Thanks again!

stewartlord commented:
Ooh nice work buddy 👍 Just checking the diffs now..

stewartlord commented on README.md:
````diff
@@ -9,18 +9,34 @@ Demo
 Usage
 -----
 ```js
+// set options
````
Would you be opposed to a basic usage example with just hash and inline size {size: 420} -- followed by the more complete usage?

ppuleo replied:
Sure, makes sense. That's probably the most common use case. I'll revise.

stewartlord commented on demo.html:
```diff
@@ -1,25 +1,87 @@
+              hashes = [
+              {id: 'stewartlord', hash: 'd6fe8c82fb0abac17a702fd2a94eff37'},
+              {id: 'jasonlong',   hash: '46384036044a604b6b3316fc167fc15f'},
+              {id: 'infusion',    hash: '373ff2375be91ebc97501e67d24ccd4d'}
```
Add your username??

ppuleo replied:
ha, ok. :smile:

stewartlord commented:
Looks great! Just a couple of minor suggestions

ppuleo commented:
ok, updated the usage section with simple/advanced examples and added my username to the demo. Good suggestions.

stewartlord commented on identicon.js:
```diff
@@ -11,29 +11,45 @@
+        this.defaults = {
+            background: [240, 240, 240, 255],
+            hash:       this.createHashFromString((new Date()).toISOString()),
+            margin:     0.10,
```
Sorry I missed this before. Can we keep the old default margin (.08)?

ppuleo replied:
oh, right. I meant to change that back after testing. Will fix.

ppuleo commented:
Restored the margin default to .08

stewartlord commented on README.md:
```diff
@@ -9,18 +9,48 @@ Demo
+* **hash** - [Optional] A unicode string that will be used to generate the image. Defaults to a random hash based on the current time.
+* **options** - [Optional] An options object used to customize the generated image.
+    * **background** - The background color expressed as an rgba value array to use for the image background. For example, use [255,0,0,255] for red. Defaults to an opaque light gray [240,240,240,255].
+    * **margin** - The decimal fraction of the size to use for margin. For example, use 0.2 for a 20% margin. Defaults to 0.1 for a 10% margin.
```
So close! Says 10% here :(

ppuleo replied:
Crap, sorry about that. Fixed.

stewartlord commented:
Thanks Phil!

ppuleo commented:
Thanks @stewartlord, much appreciated!
